src/core/namespace.c

```diff
@@ -640,7 +655,8 @@ static void drop_outside_root(const char *root_directory, MountEntry *m, size_t
 
         for (f = m, t = m; f < m + *n; f++) {
 
-                if (!path_startswith(mount_entry_path(f), root_directory)) {
+                /* ExtensionImages bases are opened in the propagate directory intentionally */
+                if (f->mode != EXTENSION_IMAGES && !path_startswith(mount_entry_path(f), root_directory)) {
```

Comment:
the debug log message this will generate is misleading, maybe handle this in a separate if block with a debug message of its own

Reply:
mmh sorry I don't grok this, the log message is printed in the same situations as before - we are simply skipping it when we process the new type now. What would be the confusion?